upup/pkg/fi/cloudup/azuretasks/vmscaleset.go

```diff
@@ -173,6 +218,9 @@ func (s *VMScaleSet) Find(c *fi.Context) (*VMScaleSet, error) {
 		StorageProfile: &VMScaleSetStorageProfile{
 			VirtualMachineScaleSetStorageProfile: profile.StorageProfile,
 		},
+		LoadBalancer: &LoadBalancer{
+			Name: to.StringPtr(loadBalancerID.LoadBalancerName),
+		},
```

Comment:
We should only be setting `LoadBalancer` if `loadBalancerID` was set above, otherwise we should keep `LoadBalancer` as its default value of nil.

Reply:
added an if statement to check if loadbalancerId is nil and if not then set LoadBalancer name.